### PR TITLE
Dto for CoursesController and an example of its mapping

### PR DESCRIPTION
**Pull request by BohdanBBG**
This PR may show unnecessary changes as this branch is the fix/dtoMapping branch where all controllers use AutoMapper.

### CODE REVIEW

File: CharlieBackend.Api/Controllers/CoursesController.cs [hunk 1]
```diff
@@ -46,27 +48,32 @@ public async Task<ActionResult> PostCourse(CreateCourseModel courseModel)
```
Comment (SerjKocherga) [inline CharlieBackend.Api/Controllers/CoursesController.cs:48]: Add to return Ok(createdCourse) above.

**Review by SerjKocherga — CHANGES_REQUESTED** [review]
``

File: CharlieBackend.Api/Controllers/CoursesController.cs [hunk 2]
```diff
@@ -46,27 +48,32 @@ public async Task<ActionResult> PostCourse(CreateCourseModel courseModel)
...
 
-            var courses = await _coursesService.GetAllCoursesAsync();
+            var courses =  await _coursesService.GetAllCoursesAsync();
+
+            var dtoResult = new List<CourseDto>();
+
+            foreach (var item in courses)
+            {
+                dtoResult.Add(_mapper.Map<CourseDto>(item));
+            }
```
Comment (whereismyroot) [inline CharlieBackend.Api/Controllers/CoursesController.cs:61]: You don't need to use mapper in controller. As suggested in previous comments, please leave only dto or model

File: CharlieBackend.Api/Controllers/CoursesController.cs [hunk 3]
```diff
@@ -4,37 +4,39 @@
...
-            var isCourseNameTaken = await _coursesService.IsCourseNameTakenAsync(courseModel.Name);
+            var isCourseNameTaken = await _coursesService.IsCourseNameTakenAsync(courseDto.Name);
 
             if (isCourseNameTaken)
             {
                 return StatusCode(409, "Course already exists!");
             }
 
-            var createdCourse = await _coursesService.CreateCourseAsync(courseModel);
+            var createdCourse = await _coursesService.CreateCourseAsync(_mapper.Map<CreateCourseModel>(courseDto));
```
Comment (whereismyroot) [inline CharlieBackend.Api/Controllers/CoursesController.cs:39]: You don't need to use mapper in controller. As suggested in previous comments, please leave only dto or model

**Review by whereismyroot — CHANGES_REQUESTED** [review]
needs some changes

**Review by SerjKocherga — APPROVED** [review]

File: CharlieBackend.Business/Services/Interfaces/ICourseService.cs [hunk 4]
```diff
@@ -1,16 +1,16 @@
...
+﻿using CharlieBackend.Core.DTO.Course;
 using System.Collections.Generic;
 using System.Threading.Tasks;
 
 namespace CharlieBackend.Business.Services.Interfaces
 {
     public interface ICourseService
     {
-        public Task<CourseModel> CreateCourseAsync(CourseModel courseModel);
+        public Task<CreateCourseDto> CreateCourseAsync(CreateCourseDto courseModel);
```
Comment (whereismyroot) [inline CharlieBackend.Business/Services/Interfaces/ICourseService.cs:9]: This method should return CourseDto

File: CharlieBackend.Business/Services/Interfaces/ICourseService.cs [hunk 5]
```diff
@@ -1,16 +1,16 @@
...
 
-        public Task<IList<CourseModel>> GetAllCoursesAsync();
+        public Task<IList<CourseDto>> GetAllCoursesAsync();
 
-        public Task<CourseModel> UpdateCourseAsync(CourseModel courseModel);
+        public Task<UpdateCourseDto> UpdateCourseAsync(long id, UpdateCourseDto courseModel);
```
Comment (whereismyroot) [inline CharlieBackend.Business/Services/Interfaces/ICourseService.cs:13]: This method should return CourseDto

File: CharlieBackend.Core/AdapterExtensionMethods.cs [hunk 6]
```diff
@@ -71,36 +69,24 @@ public static LessonModel ToLessonModel(this Lesson lesson)
...
-            };
-        }
+        //public static CourseModel ToCourseModel(this Course course)
+        //{
+        //    return new CourseModel
+        //    {
+        //        Id = course.Id,
+        //        Name = course.Name,
+        //    };
+        //}
```
Comment (whereismyroot) [inline CharlieBackend.Core/AdapterExtensionMethods.cs:89]: Please remove this

File: CharlieBackend.Core/AdapterExtensionMethods.cs [hunk 7]
```diff
@@ -71,36 +69,24 @@ public static LessonModel ToLessonModel(this Lesson lesson)
...
         }
+    
+        //public static Course ToCourse(this CourseModel courseModel)
+        //{
+        //    return new Course
+        //    {
+        //        Id = courseModel.Id,
+        //        Name = courseModel.Name
+        //    };
+        //}
```
Comment (whereismyroot) [inline CharlieBackend.Core/AdapterExtensionMethods.cs:80]: Please remove this

**Review by whereismyroot — APPROVED** [review]

**Pull request merged**